SEAMJMS-15

johnament opened the pull request:
Added ARM support for Connection and consumers that may not be built using the injection API.

johnament committed:
SEAMJMS-15 Added ARM support.

johnament committed:
SEAMJMS-3 For egress, added additional API to Route and RouteImpl.

johnament committed:
SEAMJMS-3 Added observer support for interfaces defined with methods that have @Observes.

johnament committed:
SEAMJMS-15 Added disposers for built in producers.

jganoff commented:
c12a9b1: Doesn't @Disposes already provide the functionality your @Closeable implements?

79b5701: Great catch. We definitely need to be disposing of resources we're producing.

johnament committed:
SEAMJMS-3 Fixed a bug in the test case.

johnament commented:
c12a9b1: But you can only dispose managed objects.  If the target in this case is not managed, you can't dispose it.  But you can fire it in an event.

johnament commented:
In the case of the ARM, it's application scoped as well.  So it should only be destroying once, unless there's somewhere that I didn't notice.

johnament committed:
Fixed code to work with ingress listeners.

johnament committed:
SEAMJMS-3 and SEAMJMS-4.  Added support for explicitly adding routes to only a single route type (egress vs. ingress).

johnament committed:
Modified code base to be able to support dynamic qualifiers based on feedback from Pete Muir.

johnament committed:
Moved code that was bound to request scoped to be dependent.  Fixed TCCL issue.  Added preloading support for destinations.
Note: if you run this code, you could run into the infinite message loop issue.

johnament committed:
If qualifiers are empty, set the observed qualifiers to be default.

johnament committed:
Added log message.

johnament committed:
Attempting to clean up test cases.

johnament committed:
Final code clean up for SEAMJMS-3 and SEAMJMS-4.

johnament committed:
Cleaned up tests a bit more.

johnament committed:
Added an example project that demonstrates how you can publish REST messages via JMS Events.  Demonstrates the Mapping Interfaces.

johnament committed:
Fixed module name in readme.md for example.  Renamed readme file to match current convention.

johnament committed:
Fixed final name and removed extra sections of the pom file.

johnament committed:
Cleaned up documentation a bit.  Upgraded seam parent to latest (CR1)

johnament committed:
SEAMJMS-14 Added a combined build.
SEAMJMS-19 Changed header on all code to reference ASL, as well as the assembly.

johnament committed:
Had the wrong package in combined output jar file.

johnament committed:
Removed async behavior, didn't work as expected.  Added in support for changing @Inject point to use Routing literals.  Added new test case to verify behavior.  Reduced logging levels.

johnament committed:
Removed files (really this time).
Changed session producer to create application scoped sessions.

johnament committed:
Cleaned up documentation.  Cleaned up new classes for Message creation and object generation.  Made formerly failing tests work.  Added in code to observe servlet startup events.

johnament committed:
Implementations for various test cases and refactoring of code to provide both api and impl for API classes.
